pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,127 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configV1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var notReadyTolerationSecondsPath = []string{"apiServerArguments", "default-not-ready-toleration-seconds"}
+var unreachableTolerationSecondsPath = []string{"apiServerArguments", "default-unreachable-toleration-seconds"}
+
+// ObserveNotReadyTolerationSeconds observes the value that should be set for default-not-ready-toleration-seconds
+// api server argument on the basis of provided worker latency profile from config node object.
+func ObserveNotReadyTolerationSeconds(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains default-not-ready-toleration-seconds field.
+		ret = configobserver.Pruned(ret, notReadyTolerationSecondsPath)
+	}()
+
+	nodeLister := genericListers.(configobservation.Listers).NodeLister
+	node, err := nodeLister.Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the node object we are not able to determine worker latency profile
+		return existingConfig, append(errs, err)
+	}
+
+	// read the observed value
+	var observedDefaultNotReadyTolerationSeconds string
+	switch {
+	case node.Spec.WorkerLatencyProfile == configV1.DefaultUpdateDefaultReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.DefaultNotReadyTolerationSeconds)
+	case node.Spec.WorkerLatencyProfile == configV1.MediumUpdateAverageReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.MediumNotReadyTolerationSeconds)
+	case node.Spec.WorkerLatencyProfile == configV1.LowUpdateSlowReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.LowNotReadyTolerationSeconds)
+	// in case of empty worker latency profile, do not update config
+	case node.Spec.WorkerLatencyProfile == "":
+		return existingConfig, errs
+	}
+
+	// read the current value
+	var currentDefaultNotReadyTolerationSeconds string
+	currentDefaultNotReadyTolerationSecondsSlice, _, err := unstructured.NestedStringSlice(
+		existingConfig, notReadyTolerationSecondsPath...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract not ready toleration seconds from the existing config: %v", err))
+		// keep going, we are only interested in the observed value which will overwrite the current configuration anyway
+	}
+	if len(currentDefaultNotReadyTolerationSecondsSlice) > 0 {
+		currentDefaultNotReadyTolerationSeconds = currentDefaultNotReadyTolerationSecondsSlice[0]
+	}
+
+	// see if the current and the observed value differ
+	observedConfig := map[string]interface{}{}
+	if currentDefaultNotReadyTolerationSeconds != observedDefaultNotReadyTolerationSeconds {
```

Comment:
I run into the same issue recently. It seems there is no need to compare with the current value. For simplicity, you could always write the observed value.